src/Akeneo/Pim/Enrichment/Bundle/Controller/InternalApi/Product/UpdateProductController.php

```diff
@@ -175,9 +178,10 @@ private function updateProduct(ProductInterface $product, array $data): void
         $userIntents = $handledStamp->getResult();
 
         $userId = $this->userContext->getUser()?->getId();
+        $identifierCode = $this->attributeRepository->getIdentifierCode();
         $command = UpsertProductCommand::createFromCollection(
             $userId,
-            $product->getIdentifier() ?? '',
+            ProductIdentifier::fromAttributeCodeAndIdentifier($identifierCode, $product->getIdentifier()) ?? '',
```

Comment:
```suggestion
            ProductUuid::fromUuid($product->getUuid()),
```

Using uuid is always better than using the identifier. And we don't need the AttributeRepository anymore